### PR TITLE
Add support for QLinearMul ONNX op

### PR DESCRIPTION
**Pull request by anwang2009**
as title.

### CODE REVIEW

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -3279,6 +3279,40 @@ def get_scalar(x, dtype="float32"):
...
+        b_scale = get_scalar(inputs[4])
+        b_zero_point = get_scalar(inputs[5], "int32")
+        y_scale = fold_constant(get_scalar(inputs[6]))
+        y_zero_point = get_scalar(inputs[7], "int32")
+
+        dtype = infer_type(a).checked_type.dtype
+
+        ## Onnxruntime doesn't actually do this op in integer, they dequantize to fp32
+        ## and then requantize afer
+        ## https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/mlas/lib/qlmul.cpp
```
Comment (mbrookhart) [inline python/tvm/relay/frontend/onnx.py:3309]: :scream:

**Review by mbrookhart — COMMENTED** [review]
QLinearMatMul is actually a standard ONNX op, not a onnxruntime contrib op https://github.com/onnx/onnx/blob/master/docs/Operators.md#QLinearMatMul

Can we enable the onnx tests with this solution? https://github.com/apache/tvm/blob/4216cd7f9d06a7feb712e78892061721410cc224/tests/python/frontend/onnx/test_forward.py#L4771-L4772

I worry ONNX might be expecting integer computation while ORT is doing FP32.

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (mbrookhart) [reply python/tvm/relay/frontend/onnx.py:3309]: This seems to be the ORT implementation of QLinearMul, not QLinearMatMul?

Comment (anwang2009) [reply python/tvm/relay/frontend/onnx.py:3309]: Translated everything to QLinearMul instead 🤔 , let me know if it looks right to you now

**Review by mbrookhart — COMMENTED** [review]
LGTM

**Review by junrushao1994 — APPROVED** [review]

**Review by mbrookhart — APPROVED** [review]
I didn't click the green checkmark!

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (jverma-quic) [inline python/tvm/relay/frontend/onnx.py:3309]: Even if onnxruntime is performing fp32 operations, is there any reason to do the same here? Wouldn't it be better (at least some what for the performance) to requantize both inputs 'a' and 'b' as per output scale and zero_point and then perform integer matmul?

Comment (mbrookhart) [reply python/tvm/relay/frontend/onnx.py:3309]: We have a pass (https://github.com/apache/tvm/blob/a31ebf75350946f09b2965576ba276e526f8d4ac/python/tvm/relay/transform/transform.py#L1177-L1204) To do exactly that if the user wants to accept that small accuracy difference in exchange fdor the speedup.

BTW - this is elementwise multiplication, not matmul

**Pull request merged**